### ISSUE TITLE
[android] `RuntimeException` when app gets paused and returns to foreground

### ISSUE THREAD

**Issue by haveneersrobin** [tradle/react-native-udp#196]
## Description
We see the following errors in our Sentry logging: 
```
java.lang.NullPointerException: Attempt to invoke virtual method 'void java.net.DatagramSocket.send(java.net.DatagramPacket)' on a null object reference
    at com.tradle.react.c.a(UdpSenderTask.java:3)
    at com.tradle.react.c.doInBackground(UdpSenderTask.java:1)
    at android.os.AsyncTask$2.call(AsyncTask.java:333)
    at java.util.concurrent.FutureTask.run(FutureTask.java:266)
    at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:245)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1162)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:636)
    at java.lang.Thread.run(Thread.java:764)
java.lang.RuntimeException: An error occurred while executing doInBackground()
    at android.os.AsyncTask$3.done(AsyncTask.java:353)
    at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:383)
    at java.util.concurrent.FutureTask.setException(FutureTask.java:252)
    at java.util.concurrent.FutureTask.run(FutureTask.java:271)
    at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:245)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1162)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:636)
    at java.lang.Thread.run(Thread.java:764)
```

This error has occured for multiple users with different devices (all Android). The error always occurs after the activity was paused (e.g. app put to background) and the user returns to the application.

I can not share the exact code but I believe the error occurs if the user suspends the app before a response was received for a UDP broadcast message and then returns to the app.

## Current behavior
An error is thrown

## Expected behavior
No error is thrown, null reference is handled gracefully (?)

## Relevant information
|             |     |
| ----------- | --- |
| OS | Android 8, 9, 10 and 11 (Nokia 5.4, Nokia 8, Samsung Galaxy A51, Samsung Galaxy S20) |
| react-native | 0.64 and newer   |
| react-native-udp  | 4.1.3 (only this version I believe)   |

**Comment by gibo77** [comment]
How long was the App waiting for a response before the user exited the App?

**Comment by haveneersrobin** [comment]
Not lang, the Sentry logs for the starting of the UDP broadcast and the 
```
{
screen: MainActivity, 
state: paused
}
```
event, have the same timestamp... Which makes me wonder: is it possible the app is crashing?

**Comment by eliaslecomte** [comment]
I also had a look at this and created this pr: https://github.com/tradle/react-native-udp/pull/197.

Given that it's a nullpointer exception it looks like either:
* A rogue AsyncTask, which is deprecated
* The close function has been called but there are still ongoing send packets

I've added both a safeguard and replaced the AsyncTask with a better alternative.

**Comment by Rapsssito** [comment]
Fixed by #197.
